core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java

```diff
@@ -436,14 +436,25 @@ private boolean add(Block block, boolean tryConnecting,
                 return true;
             }
 
+            final StoredBlock storedPrev;
+            final Integer height;
+            final Set<VerificationFlags> flags;
+
             // Prove the block is internally valid: hash is lower than target, etc. This only checks the block contents
             // if there is a tx sending or receiving coins using an address in one of our wallets. And those transactions
             // are only lightly verified: presence in a valid connecting block is taken as proof of validity. See the
             // article here for more details: http://code.google.com/p/bitcoinj/wiki/SecurityModel
             try {
                 block.verifyHeader();
+                storedPrev = getStoredBlockInCurrentScope(block.getPrevBlockHash());
+                if (null != storedPrev) {
```

Comment:
Can't convince you to chance, can I? Putting the constant first means accidentally typing "=" instead of "!=" or "==" fails at compilation instead of runtime.